src/core/Sampler.tsx

```diff
@@ -128,12 +128,47 @@ export const Sampler = ({
 
       dummy.updateMatrix()
 
-      instancedRef.current.setMatrixAt(i, dummy.matrix)
+      if (instanceMesh?.current) {
+        instanceMesh.current.setMatrixAt(i, dummy.matrix)
+      }
+
+      dummy.matrix.toArray(buffer.array, i * 16)
     }
 
-    instancedRef.current.instanceMatrix.needsUpdate = true
+    if (instanceMesh?.current) {
+      instanceMesh.current.instanceMatrix.needsUpdate = true
+    }
+
+    buffer.needsUpdate = true
+
+    setBuffer(buffer.clone())
+  }, [mesh, instanceMesh, weight, count, transform])
+
+  return buffer
+}
+
+export const Sampler = ({
+  children,
+  weight,
+  transform,
+  instances,
+  mesh,
+  count = 16,
+  ...props
+}: React.PropsWithChildren<Props & GroupProps>) => {
```

Comment:
This is something we could hint at via linter but you might want to prefer using named functions (e.g. `function Sampler(...) {}`) so they show up properly in browsers' profilers and react devtools. Arrow functions show up as anonymous.

```jsx
// Anonymous
const Sampler = (props) => {}
// Named
function Sampler(props) {}
// Named, with forward ref (hypothetical)
const Sampler = React.forwardRef(function Sampler(props, ref) {})
```

Reply:
I see, sure will change